xtask/codegen/yaml.ungram

```diff
@@ -30,14 +30,66 @@ YamlBogus = SyntaxElement*
 YamlBogusValue = SyntaxElement*
 
 YamlRoot =
-    bom: 'UNICODE_BOM'?
-    documents: YamlDocumentList
-    eof: 'EOF'
+	bom: 'UNICODE_BOM'?
+	documents: YamlDocumentList
+	eof: 'EOF'
 
 YamlDocumentList = YamlDocument*
-YamlDocument = content: YamlContentList
-YamlContentList = AnyYamlContent*
+YamlDocument =
+	'---'? // Document start marker
+	body: AnyYamlValue
+	'...'?
 
-AnyYamlContent = YamlScalar | YamlBogusValue
-YamlScalar = value: 'yaml_scalar'
-YamlStringLiteral = value: 'yaml_string_literal'
+
+YamlNumberValue = value: 'yaml_number_value'
+YamlStringValue = value: 'yaml_string_value'
+YamlBooleanValue = value: 'yaml_boolean_value'
+YamlNullValue = value: 'yaml_null_value'
+YamlIdentifier = value: 'yaml_identifier'
+
+AnyYamlScalar =
+	YamlStringValue
+	| YamlNumberValue
+	| YamlBooleanValue
+	| YamlNullValue
+
+YamlArrayInline =
+	'['
+	items: YamlArrayInlineList
+	']'
+
+YamlArrayInlineList = (AnyYamlScalar (',' AnyYamlScalar)* ','?)
+
+AnyYamlValue =
+	AnyYamlScalar
+	| YamlArray
+	| YamlArrayInline
+	| YamlObject
+	| YamlBogusValue
+
+YamlArray =
+	items: YamlArrayItemList
+
+YamlArrayItem =
+	'-'
+	item: AnyYamlValue
+
+YamlArrayItemList = YamlArrayItem*
+
+YamlObject =
+	members: YamlObjectMemberList
+
+YamlObjectMember =
+	key: YamlIdentifier
+	':'
+	value: AnyYamlValue
+
+YamlObjectMemberList = YamlObjectMember*
+
+YamlBlockLiteral =
+	'|'
+	value: 'yaml_block'
+
+YamlBlockFolded =
+	'>'
+	value: 'yaml_block'
```

Comment:
You should create a `YamlBlockLiteral`

Reply:
In the yaml spec, the `|` char denotes a "block literal" where newlines are included in the value, and the `>` denotes a "folded block" where newlines are ignored, so I think it should be `YamlBlockValue` instead.

Reply:
The main idea was to use the same node in both definitions. It seems they are the same